pkg/testutils/jobutils/backup_restore.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package jobutils
+
+import (
+	"context"
+	gosql "database/sql"
+	"fmt"
+
+	"github.com/cockroachdb/errors"
+)
+
+// CheckDownloadCompletenessOnDatabase checks the download job actually
+// completed over the whole database.
+func CheckDownloadCompletenessOnDatabase(
+	ctx context.Context, conn *gosql.DB, dbName string,
+) (uint64, error) {
+	var databaseID int
+	row := conn.QueryRowContext(ctx, `SELECT id FROM system.namespace WHERE name = $1`, dbName)
+	if err := row.Scan(&databaseID); err != nil {
+		return 0, errors.New("could not get database descriptor id")
+	}
+	rows, err := conn.QueryContext(ctx, `SELECT id FROM system.namespace where "parentID" = $1`, databaseID)
```

Comment:
we should avoid poking directly into system tables that would then make this test overly prescriptive about something it isn't actually interested in testing, like the namespace table when there are already available APIs for doing name to id and name to span resolution., e.g. cast to regclass and oid or crdb_internal.table_span.